### DISCUSSION TITLE
Vercel Auto cropping my image once deployed

### DISCUSSION THREAD

**Discussion by Lighterchu** [vercel/community#807] [Help] [answered]
Hi Team, 

I have a storefront where all images look fine on the local side of the project and images fit and all that good stuff. However, 
once I take it and deploy it to vercel, for some reason I found the images have been cropped and changed around. The client I am working for needs to have the whole image in the product cards and not cropped. They are a photographer that needs to have images shown to the fullest. I really hope someone knows how to stop the auto-cropping or can show me the light on this issue

( DEV Server )
<img width="914" alt="Screen Shot 2022-08-19 at 5 06 11 pm" src="https://user-images.githubusercontent.com/17910103/185741870-1e902bd8-6f22-41d0-b440-57e129ec10f5.png">

( Vercel Deploy Server )
<img width="874" alt="Screen Shot 2022-08-19 at 2 12 17 am" src="https://user-images.githubusercontent.com/17910103/185741836-99ffad89-6be1-458d-bfb7-d67110a59b98.png">

I have done alot of changes in the app and tried to work it ou myself,  but nothing I do help when it comes to images as something built into the deploy change the image scale and crops it

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @Lighterchu! Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can take a closer look? 

There are a lot of things that could affect image display so I can't say for sure without looking at it. The first thing that comes to mind is [`layout="fill"`](https://nextjs.org/docs/api-reference/next/image#layout) with `next/image`, which could result in this kind of cropping.

**Reply by Lighterchu** [reply]
Hi Amyegan, 

which part of the code would you like to see. 
I am using the framework Hydrogen for shopify, how ever I am only using Vercel for the hosting services. 

there is nothing in my project that has next/image :/ but this my code that I have :)  



The imports 

import {MediaFile} from '@shopify/hydrogen/client';
import {Image, gql} from '@shopify/hydrogen';
import type {MediaEdge} from '@shopify/hydrogen/storefront-api-types';
import {ATTR_LOADING_EAGER} from '~/lib/const';

`export function ProductGallery({
  media,
  className,
}: {
  media: MediaEdge['node'][];
  className?: string;
}) {
  if (!media.length) {
    return null;
  }

  return (
    <ul
      className={`swimlane md:grid-flow-row hiddenScroll md:p-0 md:overflow-x-auto md:grid-cols-2 ${className}`}
    >
      {media.map((med, i) => {
        let mediaProps: Record<string, any> = {};
        const isFirst = i === 0;
        const isFourth = i === 3;
        const isFullWidth = i % 3 === 0;

        const data = {
          ...med,
          image: {
            // @ts-ignore
            ...med.image,
            altText: med.alt || 'Product image',
          },
        };

        switch (med.mediaContentType) {
          case 'IMAGE':
            mediaProps = {
              width: 800,
              widths: [400, 800, 1200, 1600, 2000, 2400],
            };
            break;
          case 'VIDEO':
            mediaProps = {
              width: '100%',
              autoPlay: true,
              controls: false,
              muted: true,
              loop: true,
              preload: 'auto',
            };
            break;
          case 'EXTERNAL_VIDEO':
            mediaProps = {width: '100%'};
            break;
          case 'MODEL_3D':
            mediaProps = {
              width: '100%',
              interactionPromptThreshold: '0',
              ar: true,
              loading: ATTR_LOADING_EAGER,
              disableZoom: true,
            };
            break;
        }

        if (i === 0 && med.mediaContentType === 'IMAGE') {
          mediaProps.loading = ATTR_LOADING_EAGER;
        }

        const style = [
          isFullWidth ? 'md:col-span-2' : 'md:col-span-1',
          isFirst || isFourth ? '' : 'md:aspect-[4/5]',
          'aspect-square snap-center card-image bg-white dark:bg-contrast/10 w-mobileGallery md:w-full',
        ].join(' ');

        return (
          <ul
            // @ts-ignore
            key={med.id || med.image.id}
            className={style}
          >
            <MediaFile data={data} />;
          </ul>
        );
      })}
    </ul>
  );
}`

**Reply by amyegan** [reply]
Thanks for sharing this info! Does it still crop the image if you take out `md:aspect-[4/5]` or `aspect-square` from `style`?

**Reply by Lighterchu** [reply]
so the problem I am having, in local I have change all this and did remove the cropping, how ever when it deploys to Vercel, there is something in the back end, that is forcing the changes and cropping the image, all looks amazing on local, but once it hits the server, it all goes out the window. Did you want me to show you what it looks like on local as well :) ?

**Reply by amyegan** [reply]
Do you have a recent deployment with the `aspect-square` class removed? I'm still seeing it when I visit the URL from your repo. When I take it out in dev tools, the image does display in its original aspect ratio without cropping. The cropping is definitely being caused by CSS.

Before:
<img width="970" alt="image" src="https://user-images.githubusercontent.com/8247105/186172313-4439e210-72b1-4c90-998e-6025282e5d18.png">

After:
<img width="959" alt="image" src="https://user-images.githubusercontent.com/8247105/186172772-92807cab-3ce6-4293-8230-26856b5a73c7.png">

**Reply by Lighterchu** [reply]
Doing another build, I am sure i removed it, but I could missed it, ill see what this build does and ill send you the link

**Comment by Lighterchu** [comment]
I have also my the repo public so you can see what is happening in the code if need to
